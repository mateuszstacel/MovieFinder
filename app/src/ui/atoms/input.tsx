import React, { useState, useEffect } from "react";
import styled from "styled-components";
import { onInputChange } from "../../helper/event";

interface IComponent {
  className?: string;
  placeholder: string;
  onChange: (value: string) => void;
  inputValue: string;
}

const Component: React.FunctionComponent<IComponent> = (props: IComponent) => {
  const [inputValue, setInputValue] = useState("");

  useEffect(() => {
    props.onChange(inputValue);
  }, [inputValue]);

  return (
    <div className={props.className}>
      <div className="control has-icons-left">
        <input
          onChange={onInputChange(setInputValue)}
          className="input is-medium"
          type="text"
          value={props.inputValue}
          placeholder={props.placeholder}
        />
        <span className="icon is-left">
          <i className="fas fa-search"></i>
        </span>
      </div>
    </div>
  );
};

export const InputComponent = styled(Component)`
  .control {
    margin-left: 25%;
  }

  .is-medium {
    @media screen and (min-device-width: 300px) and (max-device-width: 845px) {
      margin-left: -5%;
    }
  }

  .icon {
    @media screen and (min-device-width: 300px) and (max-device-width: 845px) {
      margin-left: -5%;
    }
  }

  .has-icons-left {
    @media screen and (min-device-width: 360px) and (max-device-width: 845px) {
      width: 100%;
    }
  }

  .input {
    width: 70%;
  }
`;
