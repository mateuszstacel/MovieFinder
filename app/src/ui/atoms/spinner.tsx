import React from "react";
import styled from "styled-components";

interface ISpinnerProps {
  className?: string;
  label: string;
}

const SpinnerComponent: React.FunctionComponent<ISpinnerProps> = (
  props: ISpinnerProps
) => {
  return (
    <div className={props.className}>
      <span />
      <label>{props.label}</label>
    </div>
  );
};

export const Spinner = styled(SpinnerComponent)`
  display: flex;
  flex-direction: column;
  position: relative;
  justify-content: center;
  align-items: center;
  span:after {
    animation: spinAround 500ms infinite linear;
    border: 2px solid #dbdbdb;
    border-radius: 290486px;
    border-right-color: transparent;
    border-top-color: transparent;
    content: "";
    display: block;
    height: 5em;
    position: relative;
    width: 5em;
    left: calc(50% - (5em / 2));
    top: calc(50% - (5em / 2));
  }
  label {
    margin: 12px;
  }
`;
