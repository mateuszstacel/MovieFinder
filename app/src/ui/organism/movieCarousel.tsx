import React, { useState, useEffect, useLayoutEffect } from "react";
import Carousel from "react-multi-carousel";
import "react-multi-carousel/lib/styles.css";
import styled from "styled-components";
import { Link } from "react-router-dom";

interface IKidsMovies {
  id: number;
  title: string;
  poster_path: string;
}

interface IResponseKidsMovies {
  page: number;
  total_results: number;
  total_pages: number;
  results: IKidsMovies[];
}

interface IKidsdMoviesProps {
  className?: string;
  movieType?: string;
  tvType?: string;
  title?: string;
  type: string;
}

interface ITopKidsMoviesState {
  data: IKidsMovies[];
}
class PopularMoviesComponent extends React.Component<
  IKidsdMoviesProps,
  ITopKidsMoviesState
> {
  constructor(props: any) {
    super(props);
    // Don't call this.setState() here!
    this.state = { data: [] };
  }

  componentDidMount() {
    this.GetMovies();
  }

  GetMovies() {
    switch (this.props.type) {
      case "tvShow":
        fetch(
          `https://api.themoviedb.org/3/tv/${this.props.tvType}?api_key=9f471da832491516e75802f839e2bae2&language=en-US&page=1`
        )
          .then(res => {
            return res.json();
          })
          .then((response: IResponseKidsMovies) => {
            this.setState({ data: this.shuffle(response.results) });
            return;
          });
        return;
      case "movie":
        fetch(
          `https://api.themoviedb.org/3/discover/movie?${this.props.movieType}&api_key=9f471da832491516e75802f839e2bae2`
        )
          .then(res => {
            return res.json();
          })
          .then((response: IResponseKidsMovies) => {
            this.setState({ data: this.shuffle(response.results) });
            return;
          });
        return;
    }
  }

  shuffle(array: any[]) {
    let currentIndex: any = array.length,
      temporaryValue,
      randomIndex;

    // While there remain elements to shuffle...
    while (0 !== currentIndex) {
      // Pick a remaining element...
      randomIndex = Math.floor(Math.random() * currentIndex);
      currentIndex -= 1;

      // And swap it with the current element.
      temporaryValue = array[currentIndex];
      array[currentIndex] = array[randomIndex];
      array[randomIndex] = temporaryValue;
    }

    return array;
  }

  render() {
    const responsive = {
      superLargeDesktop: {
        // the naming can be any, depends on you.
        breakpoint: { max: 4000, min: 3000 },
        items: 5
      },
      desktop: {
        breakpoint: { max: 3000, min: 1024 },
        items: 4
      },
      tablet: {
        breakpoint: { max: 1024, min: 464 },
        items: 2
      },
      mobile: {
        breakpoint: { max: 464, min: 0 },
        items: 3
      }
    };

    const HandleClick = () => {
      this.setState({ ...this.state, data: this.shuffle(this.state.data) });
      window.scrollTo({
        top: 0,
        left: 0,
        behavior: "smooth"
      });
    };

    const handleRouting = (itemId: number, posterPath: string) => {
      switch (this.props.type) {
        case "tvShow":
          return (
            <Link to={`/tv-shows-library/tv-show/${itemId}`}>
              <img
                key={itemId}
                src={`https://image.tmdb.org/t/p/w500${posterPath}`}
              />
            </Link>
          );
        case "movie":
          return (
            <Link to={`/movies-library/movie/${itemId}`}>
              <img
                key={itemId}
                src={`https://image.tmdb.org/t/p/w500${posterPath}`}
              />
            </Link>
          );
      }
    };

    return (
      <div className={this.props.className}>
        <label className="Label">{this.props.title}</label>
        <Carousel
          responsive={responsive}
          swipeable={false}
          draggable={false}
          infinite={true}
          centerMode={true}
        >
          {this.state.data.map((item: IKidsMovies) => {
            return item.poster_path != undefined ? (
              <div onClick={HandleClick} key={item.id} className="Movie">
                {handleRouting(item.id, item.poster_path)}
              </div>
            ) : (
              ""
            );
          })}
        </Carousel>
      </div>
    );
  }
}

export const CarouselComponent = styled(PopularMoviesComponent)`
  font-family: "Times New Roman";
  background: black;

  .Label {
    font-size: 3vw;
    margin-left: 40px;
    color: white;
    font: bold;

    @media screen and (min-device-width: 300px) and (max-device-width: 845px) {
      font-size: 7vw;
      color: white;
      margin-left: 10px;
    }
  }
  .Movie {
    width: 250px;

    @media screen and (min-device-width: 300px) and (max-device-width: 845px) {
      width: 120px;
    }
  }
`;
