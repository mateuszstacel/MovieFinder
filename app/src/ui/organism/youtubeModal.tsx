import React from 'react'

interface IPlayMovieComponent {
    movieKey: string;
    onClose: () => void;
}
export const PlayMovieComponent: React.FunctionComponent<IPlayMovieComponent> = (props: IPlayMovieComponent) => {

    return(   

                 <div className="modal is-active">
    <div className="modal-background"></div>

    <div className="modal-card">
    <button onClick={props.onClose} className="modal-close is-large" aria-label="close"></button>
      <section className=" viedoModal">
      <iframe className="YoutubeVideo"
                       src={`https://www.youtube.com/embed/${props.movieKey}`}>
                  </iframe>}
      </section>
    </div>
  </div>





  )
}