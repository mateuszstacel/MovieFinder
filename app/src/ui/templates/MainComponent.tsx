import React, { useState } from "react";
import styled from "styled-components";
import Background from "../../../public/images/mainBackground.jpg";
import { NavBar } from "./navbar";
import { InputComponent } from "../atoms/input";
import { Title } from "../templates/title";
import { SearchViewComponent } from "../organism/searchResultModal";

interface IMainComponent {
  className?: string;
  onInputChange: () => (value: string) => void;
  currentSearch: (type: string) => void;
  inputValue: string;
}

const Component: React.FunctionComponent<IMainComponent> = (
  props: IMainComponent
) => {
  const [isLoaded, setIsLoaded] = useState(false);
  const [isMovieActive, setIsMovieActive] = useState(true);
  const [isTvShowActive, setIsTVShowActive] = useState(false);

  const HandleCategoryChange = () => {
    if (isMovieActive) {
      setIsMovieActive(false);
      setIsTVShowActive(true);
      props.currentSearch("tvShow");
    } else {
      setIsMovieActive(true);
      setIsTVShowActive(false);
      props.currentSearch("movie");
    }
  };

  return (
    <div className={props.className}>
      <NavBar
        onCategoryChange={HandleCategoryChange}
        onTvShowActive={isTvShowActive}
        onMovieActive={isMovieActive}
      />
      <br />
      <Title />
      <InputComponent
        onChange={props.onInputChange()}
        inputValue={props.inputValue}
        placeholder={isMovieActive ? "Search for Movie" : "Search for Tv Shows"}
      />
      <small className="control resSub helper has-text-white ">
        Action, Advertisement, Adventure, Comedy and much more...
      </small>
    </div>
  );
};

export const MainComponent = styled(Component)`
  width: 100%;
  height: 500px;
  background-image: url("https://i.pinimg.com/originals/0c/1b/54/0c1b541757e16d7c32736c0ec00d416f.jpg");
  background-size: 100%;
  background-repeat: no-repeat;

  @media screen and (min-device-width: 300px) and (max-device-width: 845px) {
    width: 100%;
    height: 280px;
  }

  .Tittle {
    text-align: center;
    font-size: 3vw;
    color: white;
  }
  .Logo {
    font-size: 2vw;
    font-family: "Times New Roman";
  }

  .resSub {
    @media screen and (min-device-width: 360px) and (max-device-width: 845px) {
      font-size: 2.3vw;
    }
  }

  .control {
    margin-left: 25%;
  }

  .input {
    width: 70%;
  }
`;
