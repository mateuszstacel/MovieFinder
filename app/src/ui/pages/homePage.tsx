import React, { useEffect, useState } from "react";
import styled from "styled-components";
import { MainComponent } from "../templates/MainComponent";
import { CarouselComponent } from "../organism/movieCarousel";

interface IHomePage {
  className?: string;
}

enum KindsOfMovies {
  kids = "certification_country=US&certification.lte=G&sort_by=popularity.desc",
  popular = "sort_by=popularity.desc",
  comedies = "with_genres=35&with_cast=23659&sort_by=revenue.desc",
  topRated = "certification_country=US&certification=R&sort_by=vote_average.desc"
}

enum MoviesTitles {
  kids = "Kids",
  popular = "Pupular",
  topRated = "Top Rated",
  comedies = " Science Fiction"
}

const Component: React.FunctionComponent<IHomePage> = (props: IHomePage) => {
  const [inputText, setInputText] = useState("");

  const HandleInputChange = () => {
    return (value: string) => {
      setInputText(value);
    };
  };

  return (
    <div className={props.className}>
      <MainComponent onInputChange={HandleInputChange} />
      <p className="has-text-white">{inputText}</p>
      <CarouselComponent
        title={MoviesTitles.popular}
        movieType={KindsOfMovies.popular}
      />
      <CarouselComponent
        title={MoviesTitles.kids}
        movieType={KindsOfMovies.kids}
      />
      <CarouselComponent
        title={MoviesTitles.topRated}
        movieType={KindsOfMovies.topRated}
      />
      <CarouselComponent
        title={MoviesTitles.comedies}
        movieType={KindsOfMovies.comedies}
      />
    </div>
  );
};

export const HomePage = styled(Component)``;
