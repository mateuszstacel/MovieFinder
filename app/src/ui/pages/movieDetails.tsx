import React, { useState, useEffect } from "react";
import { RouteComponentProps } from "react-router-dom";
import { NavBar } from "../templates/navbar";
import { PlayMovieComponent } from "../organism/youtubeModal";
import { CarouselComponent } from "../organism/movieCarousel";

interface IMovieVideo {
  key: string;
}

interface IMovieVideoResponse {
  results: IMovieVideo[];
}

interface IMovie {
  poster_path: string;
  title: string;
  overview: string;
  tagline: string;
  vote_average: number;
  vote_count: number;
}

let defaultData: IMovie = {
  poster_path: "",
  title: "",
  overview: "",
  tagline: "",
  vote_average: 0,
  vote_count: 0
};

type TParams = { id: string };

enum KindsOfMovies {
  drama = "with_genres=18&sort_by=vote_average.desc&vote_count.gte=10",
  best2010 = "/discover/movie?with_people=108916,7467&sort_by=popularity.desc",
  withBradPitt = "/discover/movie?with_people=287,819&sort_by=vote_average.desc",
  lowRated = "/discover/movie?sort_by=popularity.asc"
}

export const MovieDetails: React.FunctionComponent<RouteComponentProps<
  TParams,
  any,
  any
>> = ({ match }: RouteComponentProps<TParams>) => {
  const [movieKey, setMovieKey] = useState("");
  const [isVideoVisible, setIsVideoVisible] = useState(false);
  const [showVideo, setShowVideo] = useState(false);
  const [data, setData] = useState(defaultData);

  useEffect(() => {
    fetch(
      `https://api.themoviedb.org/3/movie/${match.params.id}?api_key=9f471da832491516e75802f839e2bae2`
    )
      .then(res => {
        return res.json();
      })
      .then((response: IMovie) => {
        debugger;
        console.log(response);
        setData(response);
        return;
      });

    fetch(
      `https://api.themoviedb.org/3/movie/${match.params.id}/videos?api_key=9f471da832491516e75802f839e2bae2&language=en-US`
    )
      .then(res => {
        debugger;
        return res.json();
      })
      .then((response: IMovieVideoResponse) => {
        if (response.results.length > 0) {
          setMovieKey(response.results[0].key);
          setIsVideoVisible(true);
          return;
        } else {
          setShowVideo(false);
          setIsVideoVisible(false);
          return;
        }
      });
  });

  useEffect(() => {
    if (movieKey != "") {
      setIsVideoVisible(true);
    }
  }, [movieKey]);

  const HandleVideoClose = () => {
    setShowVideo(false);
  };

  const HandleVideoClick = () => {
    setShowVideo(true);
  };
  return (
    <div>
      <NavBar />

      <div className="MovieDetailsMainComponent">
        {showVideo && (
          <PlayMovieComponent movieKey={movieKey} onClose={HandleVideoClose} />
        )}
        <div className="MovieImage">
          <img
            width="300"
            height="300"
            src={`https://image.tmdb.org/t/p/w500/${data.poster_path}`}
          />
        </div>

        <div className="Description">
          <label className="title has-text-white">{data.title}</label>
          <br />
          <p>
            <i className="far fa-star has-text-warning"></i>{" "}
            {data.vote_average.toString()} of {data.vote_count.toString()}{" "}
            voutes
          </p>
          <br />
          <label className="subtitle has-text-white">{data.tagline}</label>
          <br />
          <br /> <br /> <br />
          {data.overview}
        </div>
        {isVideoVisible && (
          <div className="PlayVideo">
            {" "}
            <p className="title has-text-white" onClick={HandleVideoClick}>
              Watch thriller
              <br />
              <i className="fab fa-youtube"></i>
            </p>{" "}
          </div>
        )}
      </div>

      <CarouselComponent
        movieType={KindsOfMovies.best2010}
        title="Explore More"
      />
      <CarouselComponent movieType={KindsOfMovies.withBradPitt} />
      <CarouselComponent movieType={KindsOfMovies.drama} />
      <CarouselComponent movieType={KindsOfMovies.lowRated} />
    </div>
  );
};
