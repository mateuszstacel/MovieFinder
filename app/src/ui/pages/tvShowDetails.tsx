import React, { useState, useEffect } from "react";
import { RouteComponentProps } from "react-router-dom";
import { NavBar } from "../templates/navbar";
import { TvShowOverview } from "../organism/tvShowOverview";
import { CarouselComponent } from "../organism/movieCarousel";

interface ITVShow {
  key: string;
}

interface ITvShowResponse {
  results: ITvShow[];
}

export interface ITvShow {
  poster_path: string;
  name: string;
  overview: string;
  tagline: string;
  vote_average: number;
  vote_count: number;
}

let defaultData: ITvShow = {
  poster_path: "",
  name: "",
  overview: "",
  tagline: "",
  vote_average: 0,
  vote_count: 0
};

type TParams = { id: string };

enum KindsOfTVShows {
  popular = "popular",
  topRated = "top_rated"
}

export const TvShowDetails: React.FunctionComponent<RouteComponentProps<
  TParams,
  any,
  any
>> = ({ match }: RouteComponentProps<TParams>) => {
  const [movieKey, setMovieKey] = useState("");
  const [isVideoVisible, setIsVideoVisible] = useState(false);
  const [showVideo, setShowVideo] = useState(false);
  const [data, setData] = useState(defaultData);

  useEffect(() => {
    fetch(
      `https://api.themoviedb.org/3/tv/${match.params.id}?api_key=9f471da832491516e75802f839e2bae2`
    )
      .then(res => {
        if (res.ok) {
          return res.json();
        }
      })
      .then((response: ITvShow) => {
        setData(response);
        return;
      });

    fetch(
      `https://api.themoviedb.org/3/tv/${match.params.id}/videos?api_key=9f471da832491516e75802f839e2bae2&language=en-US`
    )
      .then(res => {
        if (res.ok) {
          return res.json();
        }
      })
      .then((response: any) => {
        if (response != undefined) {
          if (response.results.length > 0) {
            setMovieKey(response.results[0].key);
            setIsVideoVisible(true);
            return;
          } else {
            setShowVideo(false);
            setIsVideoVisible(false);
            return;
          }
        }
      });
  }, [match.params.id]);

  useEffect(() => {
    if (movieKey != "") {
      setIsVideoVisible(true);
    }
  }, [movieKey]);

  const HandleVideoClose = () => {
    setShowVideo(false);
  };

  const HandleVideoClick = () => {
    setShowVideo(true);
  };
  return (
    <div>
      <NavBar />
      <TvShowOverview
        data={data}
        isVideoVisible={isVideoVisible}
        OnVideoClose={HandleVideoClose}
        onVideoClick={HandleVideoClick}
        showVideo={showVideo}
        movieKey={movieKey}
      />
      <CarouselComponent
        tvType={KindsOfTVShows.popular}
        title="More TV Shows"
        type="tvShow"
      />
      <CarouselComponent type="tvShow" tvType={KindsOfTVShows.topRated} />
    </div>
  );
};
