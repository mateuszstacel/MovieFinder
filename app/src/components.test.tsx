import React from "react";
import { InputComponent } from "./ui/atoms/input";
import { Spinner } from "./ui/atoms/spinner";
import { CarouselComponent } from "../src/ui/organism/movieCarousel";
import { PlayMovieComponent } from "../src/ui/organism/youtubeModal";
import { MainComponent } from "./ui/templates/MainComponent";
import { Movies } from "../src/ui/templates/movies";
import { NavBar } from "../src/ui/templates/navbar";
import { Title } from "./ui/templates/title";
import { TvShows } from "../src/ui/templates/tvShows";
import { HomePage } from "../src/ui/pages/homePage";
import { MovieDetails } from "../src/ui/pages/movieDetails";
import { TvShowDetails } from "../src/ui/pages/tvShowDetails";
import { render } from "@testing-library/react";

interface IComponentUnitTests<T = any> {
  [key: string]: T;
}

const allComponents: IComponentUnitTests = {
  InputComponent,
  Spinner,
  CarouselComponent,
  PlayMovieComponent,
  MainComponent,
  Movies,
  NavBar,
  Title,
  TvShows,
  HomePage,
  MovieDetails,
  TvShowDetails
};

/**
 * Runs a basic test for each component
 */
Object.keys(allComponents).forEach(componentName => {
  describe(`<${componentName} />`, () => {
    it(`renders ${componentName} without crashing`, () => {
      const Component: any = allComponents[componentName];
    });
  });
});
