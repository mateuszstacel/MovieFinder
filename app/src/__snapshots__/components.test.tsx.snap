// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CarouselComponent /> renders CarouselComponent without crashing 1`] = `
<div>
  <div
    class="sc-htpNat bgzSJY"
  >
    <label
      class="Label"
    />
    <div
      class="react-multi-carousel-list  "
    >
      <ul
        class="react-multi-carousel-track "
        style="transition: none; overflow: unset; transform: translate3d(0px,0,0);"
      />
    </div>
  </div>
</div>
`;

exports[`<PlayMovieComponent /> renders PlayMovieComponent without crashing 1`] = `
<div>
  <div
    class="modal is-active"
  >
    <div
      class="modal-background"
    />
    <div
      class="modal-card"
    >
      <button
        aria-label="close"
        class="modal-close is-large"
      />
      <section
        class=" viedoModal"
      >
        <iframe
          class="YoutubeVideo"
          src="https://www.youtube.com/embed/undefined"
        />
        }
      </section>
    </div>
  </div>
</div>
`;

exports[`<Spinner /> renders Spinner without crashing 1`] = `
<div>
  <div
    class="sc-bwzfXH kfnZfo"
  >
    <span />
    <label />
  </div>
</div>
`;

exports[`<Title /> renders Title without crashing 1`] = `
<div>
  <div
    class="sc-ifAKCX flzyHI"
  >
    <div
      class="Tittle"
    >
      <label
        class="TittleLabel"
      >
        Your public library to find yours favorite movies.
      </label>
      <br />
      <label
        class="subtitle subtitleLabel"
      >
        Discover over million videos shared by our generous community.
      </label>
    </div>
    <br />
  </div>
</div>
`;
